### ISSUE TITLE
Example for interop with koa / express

### ISSUE THREAD

**Issue by creatorrr** [francisrstokes/hexnut#11]
Great work on Hexnut! In your blog post, you mentioned that it was possible to interop with existing projects using koajs or expressjs. Can you please share an example on how that could be done?

**Comment by creatorrr** [comment]
For reference, I am currently doing the following to get koa.js and hexnut to work on the same server:

```javascript
class CustomHexnut extends hexnut {
  startWithServer(server) {
    if (!server) {
      throw new Error("Server instance required");
    }

    this.server = new Websocket.Server({ server });
    this.isRunning = true;

    this.server.on("connection", (ws, req) => {
      const id = uuid();
      const ctx = createCtx(ws, req, this);
      this.connections[id] = ctx;

      this.runMiddleware(ctx);

      ws.on("message", message => {
        ctx._reset(message);
        this.runMiddleware(ctx);
      });

      ws.on("close", () => {
        ctx.message = null;
        ctx.type = "closing";
        this.runMiddleware(ctx).then(() => {
          delete this.connections[id];
        });
      });
    });
  }
}

export const websocketApp = new CustomHexnut({ port: env.PORT });

export const app = new Koa();
export const server = http.createServer(app.callback());

websocket.startWithServer(server);

```

**Comment by francisrstokes** [comment]
Hi there,
If you pass an http(s) server to the config object of Hexnut, the underlying ws connection will use that server. No need to subclass!

**Comment by creatorrr** [comment]
Gotcha. Thanks! Would be helpful as an addition to the docs too.

**Comment by francisrstokes** [comment]
Updated the docs here: https://github.com/francisrstokes/hexnut/blob/master/docs/api.md

Though I hope to overhaul the whole documentation at some stage. Thanks again for the feedback @creatorrr

**Comment by creatorrr** [comment]
@francisrstokes tiny bug right now, passing a `server` object to the constructor has no effect since a `port` (merged from ./config) takes precedence. So I get a port bind error:

`Error: listen EADDRINUSE: address already in use :::5000`

Possible fixes:
 - `wsConfig || config` in `Hexnut` constructor.
 - Check for `server` key in `wsConfig` and remove `host` and `port` values from the config object

What do you think? Happy to make a PR if you'd like.

**Comment by francisrstokes** [comment]
Yes true! I've already fixed and patched (0.1.1), using essentially the `wsConfig || config` solution.
